### DISCUSSION TITLE
File Upload bug?

### DISCUSSION THREAD

**Discussion by c-gross** [orchidsoftware/platform#2368] [General]
Hi,

I'm using Orchid 13.1.0 and I wrote an CSV file upload for an data import. The upload and data import ist working but I found a strange bug. I'm not sure if it's orchid or laravel related.

Once the import is done, I delete the csv file from the storage folder. If I try to upload the same file again it will not save the file in the storage folder anymore. If I change the filename it's the same issue. If I change 1 bit in the file itself, like adding a whitespace at the end, it will work again. Deleting the laravel cache with artisan cache:clear does not help.

At some point is maybe some hash/checksum value saved and is expecting that the file is still there and an new upload is not required.

I added a button with ModalToggle to the command bar:
```
ModalToggle::make('CSV Import')
                ->modal('csvUpload')
                ->method('upload')
```

The ModalToggle:
```
Layout::modal('csvUpload',
                Layout::rows([
                    Upload::make('upload')
                        ->title('Upload CSV')
                        ->maxFiles(1)
                        ->storage('local')
                        ->acceptedFiles('.csv')
                        ->required()
                ])
            )->title('Import CSV')
```

And the method for the upload:
```
public function upload(Request $request)
    {
        $attachment = Attachment::find($request->get('upload')[0]);
        $path = $attachment->physicalPath();

        if(!Storage::disk('local')->exists($path)) {
            Alert::error('Failed to open csv file');
            return;
        }

        $file = Storage::disk('local')->path($path);

        // Import logic -> after the import I delete the uploaded file from the storage

        Storage::disk('local')->delete($path);
    }
```

**Comment by tabuna** [comment]
When you upload a file to the server using the Upload/Image/Cropper.  Then for each file, a model containing the hash of the file is also created.  This avoids duplication of the same documents in the repository.

When deleting a model (even a copy).  There is a check and if this hash is used anywhere else.  If the result is positive, both the file itself and its model in the database are deleted.
